scaling_performance/optimizing_storage.adoc

```diff
@@ -11,25 +11,185 @@
+* The storage technology must ensure read-after-write consistency. All NAS storage (excluding CNS/CRS GlusterFS as it uses an object storage interface) are not
+recommended for {product-title} Registry cluster deployment with production workloads.
+* While hostPath volumes are configurable for a non-scaled/HA {product-title} Registry cluster deployment, they are not recommended.
+
+[WARNING]
+====
+Corruption may occur when using NFS to back {product-title} registry at production workloads.
+====
+
+[[scaled-registry]]
+===== Scaled Registry
+
+In a scaled/HA {product-title} registry cluster deployment:
+
+* The preferred storage technology is object storage. The storage technology must support RWX access mode and must ensure read-after-write consistency.
+* File storage and block storage are expressly not recommended for a scaled/HA {product-title} registry
+cluster deployment with production workloads.
+* All NAS storage (excluding CNS/CRS GlusterFS as it uses an object storage interface) are
+not recommended for {product-title} Registry cluster deployment with production workloads.
+
+[WARNING]
+====
+Corruption may occur when using NFS to back {production-title} scaled/HA registry at production workloads.
+====
+
+[[metrics]]
+===== Metrics
+
+In an {product-title} hosted metrics cluster deployment:
+
+* The preferred storage technology is block storage.
+* It is not recommended to use all NAS storage (excluding CNS/CRS GlusterFS as it uses a block storage interface from iSCSI) for hosted metrics cluster deployment with production workloads.
+
+[WARNING]
+====
+Corruption may occur when using NFS to back hosted metrics at production workloads.
+====
+
+[[logging]]
+===== Logging
+
+In an {product-title} hosted logging cluster deployment:
+
+* The preferred storage technology is block storage.
+* It is not recommended to use all NAS storage (excluding CNS/CRS GlusterFS as it uses a block storage interface from iSCSI) for hosted metrics cluster deployment with production workloads.
+
+[WARNING]
+====
+Corruption may occur when using NFS to back hosted logging at production workloads.
+====
+
+[[applications]]
+===== Applications
+
+Application use cases vary from application to application:
+
+* Storage technologies that support dynamic PV provisioning have low mount time latencies, and are not tied
+to nodes to support a healthy cluster.
+* NFS does not guarantee read-after-write consistency and is not recommended for applications which require it.
+* Applications that depend on writing to the same, shared NFS export may experience issues with production workloads.
+
+[[other-storage-recommendations]]
+==== Other Specific Application Storage Recommendations
+
+* {product-title} Internal etcd: For the best etcd reliability, the lowest consistent latency storage technology is preferable.
+* OpenStack Cinder: OpenStack Cinder tends to be adept in ROX access mode use cases.
```

Comment:
what is ROX

Reply:
ReadOnlyMany [0][1].

[0] https://docs.openshift.com/container-platform/3.6/architecture/additional_concepts/storage.html#pv-access-modes
[1] https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes

Reply:
We should make that clear on the first usage. Always spell it out the first time.